val.py

```diff
@@ -302,108 +322,113 @@ def run(
             if single_cls:
                 pred[:, 5] = 0
             predn = pred.clone()
+            predntwee = pred.clone()  # TODO can we use predn, or are these values changed?
             scale_boxes(im[si].shape[1:], predn[:, :4], shape, shapes[si][1])  # native-space pred
 
             # Evaluate
-            if nl:
-                tbox = xywh2xyxy(labels[:, 1:5])  # target boxes
-                scale_boxes(im[si].shape[1:], tbox, shape, shapes[si][1])  # native-space labels
-                labelsn = torch.cat((labels[:, 0:1], tbox), 1)  # native-space labels
-                correct = process_batch(predn, labelsn, iouv)
-                if plots:
-                    if tagged_data:
-                        confusion_matrix.process_batch(predn, labelsn, gt_boxes, tagged_labels)
-                    else:
+            if not production:
+                if nl:
+                    tbox = xywh2xyxy(labels[:, 1:5])  # target boxes
+                    scale_boxes(im[si].shape[1:], tbox, shape, shapes[si][1])  # native-space labels
+                    labelsn = torch.cat((labels[:, 0:1], tbox), 1)  # native-space labels
+                    correct = process_batch(predn, labelsn, iouv)
+                    if plots:
                         confusion_matrix.process_batch(predn, labelsn)
-            stats.append((correct, pred[:, 4], pred[:, 5], labels[:, 0]))  # (correct, conf, pcls, tcls)
+                stats.append((correct, pred[:, 4], pred[:, 5], labels[:, 0]))  # (correct, conf, pcls, tcls)
 
             # Save/log
             if save_txt:
                 if tagged_data:
                     save_one_txt_and_one_json(predn,
                                               save_conf,
                                               shape,
-                                              file=save_dir / 'labels' / f'{path.stem}.txt',
-                                              json_file=save_dir / 'labels_tagged' / f'{path.stem}.json',
+                                              file=txt_path / f'{path.stem}.txt',
+                                              json_file=save_dir / 'labels_tagged' / f'{path.stem}.json', # TODO, does not work in Azure
                                               confusion_matrix=confusion_matrix)
                 else:
-                    save_one_txt(predn, save_conf, shape, file=save_dir / 'labels' / f'{path.stem}.txt')
+                    save_one_txt(predn, save_conf, shape, file=txt_path / f'{path.stem}.txt')
```

Comment:
this is removing the original functionality. 
I think we still want to save labels in the `labels` folder when we run simple validation

Reply:
ah, ignore this comment, I saw that txt_path contains labels